dkg/dkg.go

```diff
@@ -172,6 +174,19 @@ func Run(ctx context.Context, conf Config) (err error) {
 
 	ex := newExchanger(tcpNode, nodeIdx.PeerIdx, peerIds, def.NumValidators)
 
+	bcastCallbacks := map[string]bcast.Callback{}
```

Comment:
what about adding support to bcast to register callbacks `(b bcast) RegisterCallback(msgID string, callBack)`?

Reply:
Wouldn't this alter the interface of `bcast` too much?

It would render the `allowedMsgIDs` argument to `bcast.New` redundant.

I'm fine with the refactor if you're okay with changing how `bcast` works.